python/pyspark/sql/tests/test_arrow.py

```diff
@@ -18,10 +18,12 @@
 import datetime
 import os
 import threading
+import calendar
 import time
 import unittest
 from typing import cast
 from collections import namedtuple
+from zoneinfo import ZoneInfo
```

Comment:
Oh, `branch-3.5` PyPy 3.8 seems to complain.
- https://github.com/apache/spark/actions/runs/8077039915/job/22066597081
```
Traceback (most recent call last):
  File "/usr/local/pypy/pypy3.8/lib/pypy3.8/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/pypy/pypy3.8/lib/pypy3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/__w/spark/spark/python/pyspark/sql/tests/test_arrow.py", line 28, in <module>
    from zoneinfo import ZoneInfo
ModuleNotFoundError: No module named 'zoneinfo'
```

Reply:
`zoneinfo` is added at Python 3.9.

![Screenshot 2024-02-29 at 21 06 59](https://github.com/apache/spark/assets/9700541/059737a9-5740-4ab8-adbd-febceb468222)

Reply:
oh I fixed this.

Reply:
Maybe, shall we ignore `zoneinfo` related `import` and tests in Python 3.8 and lower, @HyukjinKwon ?

Reply:
oh

Reply:
branch-3.5. I will backport

Reply:
Thank you!